src/server/src/main/java/io/cassandrareaper/service/RepairRunService.java

```diff
@@ -168,6 +168,10 @@ List<Segment> generateSegments(
       Map<List<String>, List<RingRange>> replicasToRange = buildReplicasToRangeMap(rangeToEndpoint);
       String cassandraVersion = clusterFacade.getCassandraVersion(targetCluster);
 
+      if (segmentCountPerNode == 0 && endpointToRange.isEmpty()) {
```

Comment:
Is there a relationship between the fact that `segmentCountPerNode == 0` and the fact that the endpoint to range mapping is empty?
My assumption would be that the segment count value should be unrelated to the failure.

Instead of adding a conditional, would it be possible to catch the `IllegalArgumentException` in the surrounding try, and rethrow it as a `ReaperException`? This would allow to leverage the existing Precondition check from `computeGlobalSegmentCount`.

Reply:
Thanks for reviewing - I made the changes.